packages/core/src/browser/messaging/ws-connection-provider.ts

```diff
@@ -52,12 +57,12 @@ export class WebSocketConnectionProvider extends AbstractConnectionProvider<WebS
         const socket = this.createWebSocket(url);
         socket.onerror = console.error;
         socket.onopen = () => {
-            this.onSocketOpenedEmitter.fire(undefined);
+            this.fireSocketDidOpen();
         };
         socket.onclose = ({ code, reason }) => {
-            this.onSocketClosedEmitter.fire(undefined);
             for (const channel of [...this.channels.values()]) {
                 channel.close(code, reason);
+                this.fireSocketDidClose({ code: code, reason: reason });
```

Comment:
For clarification:

This is what you had initially in the changeset:
```ts
socket.onclose = ({ code, reason }) => {
            this.onSocketClosedEmitter.fire(undefined);
            for (const channel of [...this.channels.values()]) {
                channel.close(code, reason);
            }
        };
```

This is what I proposed:
```ts
socket.onclose = ({ code, reason }) => {
            for (const channel of [...this.channels.values()]) {
                channel.close(code, reason);
            }
            this.onSocketClosedEmitter.fire(undefined); // <--- call the event  when the channels are closed.
        };
```

I did not mean to see an event for each service. Sorry for the confusion. Let me know if it's still not clear what I said. Thanks!

Reply:
No problem, it is easy to fix. I've pushed the new changes related to this.